src/sortStudents.ts

```diff
@@ -1,16 +1,71 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AvrageGrade
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[], sortBy: SortType, order: SortOrder,
+): Student[] {
+  const copy = [...students];
+  const avarage = (grades: number[]): number => {
+    return grades
+      .reduce((prev: number, elem: number) => prev + elem, 0) / grades.length;
+  };
+
+  switch (sortBy) {
+    case SortType.Name:
+      copy.sort((person1: Student, person2: Student) => {
+        return order === 'asc'
+          ? person1.name.localeCompare(person2.name)
+          : person2.name.localeCompare(person1.name);
+      });
+      break;
+
+    case SortType.Surname:
+      copy.sort((person1: Student, person2: Student) => {
+        return order === 'asc'
+          ? person1.surname.localeCompare(person2.surname)
+          : person2.surname.localeCompare(person1.surname);
+      });
+      break;
+
+    case SortType.Age:
+      copy.sort((person1: Student, person2: Student) => {
+        return order === 'asc'
+          ? person1.age - person2.age
+          : person2.age - person1.age;
```

Comment:
its more a comment than something to correct.
I think you could use the function parameter here insteaed of hardcode. This enum seems like it was created exactly for that reason.
_this aplies for all the other occurencies_
```suggestion
          ? person1[sortBy] - person2[sortBy]
          : person2[sortBy] - person1[sortBy];
```